packages/web/src/components/apps/workspace/editor/editor.tsx

```diff
@@ -9,24 +9,31 @@ import { useFiles } from '../../use-files';
 import { AppType, FileType } from '@srcbook/shared';
 import EditorHeader from './header';
 import { extname } from '../../lib/path';
+import { useEffect, useState } from 'react';
 import { Preview } from '../preview';
-import { useHeaderTab } from '../../use-header-tab';
 import { cn } from '@/lib/utils.ts';
 
 type PropsType = {
   app: AppType;
 };
 
 export function Editor(props: PropsType) {
-  const { tab, switchTab } = useHeaderTab();
+  const [tab, setTab] = useState<'code' | 'preview'>('code');
   const { openedFile, updateFile } = useFiles();
 
+  useEffect(() => {
+    if (!openedFile) {
+      return;
+    }
+    setTab('code');
+  }, [openedFile]);
+
   return (
-    <div className="flex flex-col">
+    <div className="grow shrink flex flex-col">
```

Comment:
Curious: what does `grow shrink` do?

Reply:
Makes the container that wraps the editor grow/shrink to take up all space in the flex layout. This became necessary due to the statusbar at the bottom making the layout a bit more complex.